### PR TITLE
Use `pip-compile-multi` option `--autoresolve`

### PR DESCRIPTION
**Pull request by nvaytet**
We use the `--autoresolve` option of `pip-compile-multi` to make sure we don't run into conflicts across multiple txt files.

According to the docs:

> This feature works only if your project has a single requirements file, that references (directly or indirectly) all other files.

So I added the `all.in` file.

TBH, I don't really understand why this is needed, I thought the whole point of `pip-compile-multi` was to make sure that the versions were the same for all envs, but anyway.

### CODE REVIEW

**Comment by nvaytet** [comment]
It seems that `virtualenv` now supports `platformdirs4` (see https://github.com/pypa/virtualenv/pull/2664), so it now resolves fine without pinning to `<3`.

The issue with `prompt-toolkit` remains, but gets resolved by `--autoresolve`.

File: template/requirements/all.in
Comment (jl-wynen) [inline template/requirements/all.in:1]: Can you add a comment about the purpose of this file? We should communicate that nobody should make an env from it.

**Review by jl-wynen — APPROVED** [review]

File: template/requirements/all.in
Comment (nvaytet) [reply template/requirements/all.in:1]: Maybe that `all.in` file should actually be generated from the list of files in the `requirements` folder?

Comment (jl-wynen) [reply template/requirements/all.in:1]: Good point!

**Review by jl-wynen — CHANGES_REQUESTED** [review]

File: template/requirements/make_deps.py [hunk 1]
```diff
@@ -75,6 +75,17 @@ def as_nightly(repo: str) -> str:
...
+    f.write("# NO ENVIRONMENT SHOULD BE CREATED FROM THIS FILE.\n")
+    f.write("#\n")
+    f.write(
+        "# We want to use the --autoresolve option of pip-compile-multi to make sure "
+        "we do not run into conflicts across multiple txt files.\n"
+    )
+    f.write(
+        "# This feature works only if the project has a single requirements file that "
+        "references (directly or indirectly) all other files.\n\n"
+    )
```
Comment (jl-wynen) [inline template/requirements/make_deps.py:89]: Minor: This would be more readable with a single multiline string.

**Comment by nvaytet** [comment]
@jl-wynen this doesn't work with the `nightly.in` files.
It will then use e.g. `scipp @ https://github.com/scipp/scipp/releases/download/nightly/scipp-nightly-cp39-cp39-manylinux_2_17_x86_64.manylinux2014_x86_64.whl` in all `.txt` files.

So I think we need to go back to pinning the `prompt-toolkit` version :-(

**Pull request merged**